pkg/cloud/tags/tags.go

```diff
@@ -63,9 +63,6 @@ func New(params *infrav1.BuildParams, opts ...BuilderOption) *Builder {
 
 // Apply tags a resource with tags including the cluster tag.
 func (b *Builder) Apply() error {
-	if b.params == nil {
```

Comment:
Let's keep this check here too, `applyFunc()` that this method calls dereference params (the check should actually be there but that method does not return error).
>  tags := infrav1.Build(*params)

Does not matter if the current caller of this method (Ensure()) has this check. If it is called somewhere else with nil params, will panic.